packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -195,12 +194,7 @@ export class ProjectActions {
   async addProject (args: AddProject) {
     const projectRoot = await this.getDirectoryPath(args.path)
 
-    const found = this.projects.find((x) => x.projectRoot === projectRoot)
-
-    if (!found) {
-      this.projects.push({ projectRoot })
-      await this.api.insertProjectToCache(projectRoot)
-    }
+    await this.updateProjectList(() => this.api.insertProjectToCache(projectRoot))
```

Comment:
We'll call update twice if args.open is true (here and within setCurrentProject). Might be worth moving this into an else below and save us a few cycles?

Reply:
Good catch, added in [f2984b8](https://github.com/cypress-io/cypress/pull/21589/commits/f2984b8c598476a444c7212bac28d84f1ad663a9)